### PR TITLE
Updating semantic doc

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
check here for [view](https://github.com/GoogleCloudPlatform/gcsfuse/blob/7086eb8c4a2084c88e22820e50fe1d54d9aed403/docs/semantics.md)

### Testing
N/A

### CODE REVIEW

File: docs/semantics.md [hunk 1]
```diff
@@ -1,563 +1,285 @@
...
 
-GCS offers no way to delete an object if and only if other objects don't exist.
-It is therefore impossible to atomically check whether a directory is empty and
-delete its backing object.
+**Pubsub notifications on file creation**
 
-gcsfuse does the pragmatic thing here: it lists objects with the directory's
-name as a prefix, returning `ENOTEMPTY` if anything shows up, and otherwise
-deletes the backing object.
+[Pubsub notifications](https://cloud.google.com/storage/docs/reporting-changes) may be enabled on a Cloud Storage bucket to help track changes to Cloud Storage objects. Due of the semantics that Cloud Storage FUSE uses to create files, 3 different events are generated, per file created:
```
Comment (raj-prince) [inline docs/semantics.md:152]: Due of -> Due to

File: docs/semantics.md [hunk 2]
```diff
@@ -1,563 +1,285 @@
...
-    in terms of GCS operations, and for large directories would have high
-    probability of failure, leaving the two directories in an inconsistent
-    state. 
+See the notes on [fuseops.FlushFileOp](http://godoc.org/github.com/jacobsa/fuse/fuseops#FlushFileOp) for more details.
 
-    *  However, if your application can tolerate the risks, you may enable
-       renaming directories in a non-atomic way in gcsfuse starting v0.35.0,
-       by setting `--rename-dir-limit`. If a directory contains fewer files
-       than this limit and no subdirectory, it can be renamed.
+Error Handling
```
Comment (raj-prince) [inline docs/semantics.md:273]: Please mark "Error Handling" also as bold as - Memory-mapped files.

**Review by raj-prince — APPROVED** [review]
Have added few comments please incorporate before merging this.

File: docs/semantics.md [hunk 3]
```diff
@@ -1,563 +1,285 @@
...
+Cloud Storage FUSE directory inodes exist simply to satisfy the kernel and export a way to look up child inodes. Unlike file inodes:
+- There are no guarantees about stability of directory inode IDs. They may change from lookup to lookup even if nothing has changed in the Cloud Storage bucket. They may not change even if the directory object in the bucket has been overwritten.
+- Cloud Storage FUSE does not keep track of modification time for directories. There are no guarantees for the contents of stat::st_mtim or equivalent, or the behavior of utimes(2) and similar.
+- There are no guarantees about stat::st_nlink.
 
+Despite no guarantees about the actual times for directories, their time fields in stat structs will be set to something reasonable.
 
-<a name="surprising-behaviors"></a>
-# Surprising behaviors
+**Unlinking**
```
Comment (raj-prince) [inline docs/semantics.md:208]: New line after this.

Comment (Tulsishah) [reply docs/semantics.md:208]: fixed

File: docs/semantics.md (hunk 2 above)
Comment (Tulsishah) [reply docs/semantics.md]: fixed

File: docs/semantics.md [hunk 4]
```diff
@@ -1,563 +1,285 @@
...
-*   Reading or modifying a file backed by an object with a `contentEncoding`
-    property set may yield surprising results, and no guarantees are made about
-    the behavior. It is recommended that you do not use gcsfuse to interact
-    with such objects.
+**Writes**
 
-    (See the [Object resource][] and [performance tips][] pages for more info
-    on `contentEncoding`, and [this][encoding-writeup] writeup for an
-    explanation of why it can't be reliably supported.)
+For modifications to existing objects, Cloud Storage FUSE downloads the entire backing object's contents from Cloud Storage. The contents are stored in a local temporary file whose location is controlled by the flag ```--temp-dir```. Later, when the file is closed or fsync'd, Cloud Storage FUSE writes the contents of the local file back to Cloud Storage as a new object generation. Modifying even a single bit of an object results in the full re-upload of the object. The exception is if an append is done to the end of a file, where the original file is at leat 2MB,  then only the appended content is uploaded.
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply docs/semantics.md]: fixed

File: docs/semantics.md (hunk 1 above)
Comment (Tulsishah) [reply docs/semantics.md]: fixed

**Comment by Tulsishah** [comment]
> Have added few comments please incorporate before merging this.

Fixed

**Pull request merged**